go/engine/pgp_pull.go

```diff
@@ -160,19 +210,26 @@ func (e *PGPPullEngine) Run(ctx *Context) error {
 			continue
 		}
 
-		for _, bundle := range user.GetActivePGPKeys(false) {
-			// Check each key against the tracked set.
-			if !trackedFingerprints[bundle.GetFingerprint().String()] {
-				ctx.LogUI.Warning("Keybase says that %s owns key %s, but you have not tracked this fingerprint before.", user.GetName(), bundle.GetFingerprint())
-				continue
-			}
+		if err = e.exportKeysToGPG(ctx, user, trackedFingerprints); err != nil {
+			return err
+		}
+	}
+	return nil
+}
 
-			err = gpgClient.ExportKey(*bundle)
-			if err != nil {
-				return err
-			}
-			ctx.LogUI.Info("Imported key for %s.", user.GetName())
+func (e *PGPPullEngine) exportKeysToGPG(ctx *Context, user *libkb.User, tfp map[string]bool) error {
+	for _, bundle := range user.GetActivePGPKeys(false) {
+		// Check each key against the tracked set.
+		if tfp != nil && !tfp[bundle.GetFingerprint().String()] {
+			ctx.LogUI.Warning("Keybase says that %s owns key %s, but you have not tracked this fingerprint before.", user.GetName(), bundle.GetFingerprint())
```

Comment:
Old code, but I'm wondering if this warning makes sense. We really should be doing the right thing with each user's sigchain, and checking the proofs that they own these keys. If that means the "fetch everyone" version takes forever, so be it. How often could you possibly need to run that command, that you can't let it sit in the background for a few minutes?